test/bindings/validator.spec.ts

```diff
@@ -505,10 +505,880 @@ test.group('Validator | exists', (group) => {
     }
   })
 
-  test('perform case-insensitive query', async ({ assert }) => {
+  test('add or where constraints', async ({ assert }) => {
+    assert.plan(2)
+
+    const [row] = await db
+      .table<{ id: string } | number>('users')
+      .returning('id')
+      .insert({ email: 'virk@adonisjs.com', username: 'virk' })
+
+    db.connection()
+      .getReadClient()
+      .on('query', ({ sql, bindings }) => {
+        const { sql: knexSql, bindings: knexBindings } = db
+          .connection()
+          .getReadClient()
+          .select(1)
+          .from('users')
+          .where('id', typeof row === 'number' ? row : row.id)
+          .where('username', 'nikk')
+          .orWhere('points', 1)
```

Comment:
Alright, so I am not sure if this is how `orWhere` should be applied. Because as per this query, we will either get the row with `(id = 1 & username = 'nikk')` or we will get a row with `points = 1`.

And think this is going to be problematic in many ways. What if someone wants to write multiple groups of where conditions. Like `(where a = 'b' & foo = 'bar')` or `(where a = 'c' & foo = 'bar')`.

I think, we will have to expose the actual underlying query builder instance for more advance use cases. However, that is something not possible as of today, because of the way schemas are cached